docs/source/java/substrait.rst

```diff
@@ -148,297 +136,60 @@ This Java program:
-            TypeCreator.NULLABLE.STRING
-        );
-        NamedStruct of = NamedStruct.of(
-            columnNames,
-            Type.Struct.builder().fields(dataTypes).nullable(false).build()
-        );
-        // Extensions URI
-        HashMap<String, SimpleExtensionURI> extensionUris = new HashMap<>();
-        extensionUris.put(
-            "key-001",
-            SimpleExtensionURI.newBuilder()
-                .setExtensionUriAnchor(1)
-                .setUri("/functions_comparison.yaml")
-                .build()
-        );
-        // Extensions
-        ArrayList<SimpleExtensionDeclaration> extensions = new ArrayList<>();
-        SimpleExtensionDeclaration extensionFunctionLowerThan = SimpleExtensionDeclaration.newBuilder()
-            .setExtensionFunction(
-                SimpleExtensionDeclaration.ExtensionFunction.newBuilder()
-                    .setFunctionAnchor(1)
-                    .setName("gt:any_any")
-                    .setExtensionUriReference(1))
-            .build();
-        extensions.add(extensionFunctionLowerThan);
-        // Extended Expression
-        ExtendedExpression.Builder extendedExpressionBuilder =
-            ExtendedExpression.newBuilder().
-                addReferredExpr(0,
-                    expressionReferenceBuilderFilterOne).
-                setBaseSchema(of.toProto(new TypeProtoConverter(
-                    new ExtensionCollector())));
-        extendedExpressionBuilder.addAllExtensionUris(extensionUris.values());
-        extendedExpressionBuilder.addAllExtensions(extensions);
-        ExtendedExpression extendedExpression = extendedExpressionBuilder.build();
-        byte[] extendedExpressions = Base64.getDecoder().decode(
-            Base64.getEncoder().encodeToString(
-                extendedExpression.toByteArray()));
-        ByteBuffer substraitExpressionFilter = ByteBuffer.allocateDirect(
-            extendedExpressions.length);
-        substraitExpressionFilter.put(extendedExpressions);
-        return substraitExpressionFilter;
+      private static ByteBuffer getByteBuffer(String[] sqlExpression) throws SqlParseException {
+        String schema =
+            "CREATE TABLE NATION (N_NATIONKEY INT NOT NULL, N_NAME VARCHAR, "
+                + "N_REGIONKEY INT NOT NULL, N_COMMENT VARCHAR)";
+        SqlExpressionToSubstrait expressionToSubstrait = new SqlExpressionToSubstrait();
+        ExtendedExpression expression =
+            expressionToSubstrait.convert(sqlExpression, ImmutableList.of(schema));
+        byte[] expressionToByte =
+            Base64.getDecoder().decode(Base64.getEncoder().encodeToString(expression.toByteArray()));
+        ByteBuffer byteBuffer = ByteBuffer.allocateDirect(expressionToByte.length);
+        byteBuffer.put(expressionToByte);
+        return byteBuffer;
       }
     }
 
 .. code-block:: text
 
-    ADD_TEN_TO_COLUMN_N_REGIONKEY	CONCAT_COLUMNS_N_NAME_AND_N_COMMENT
+    column-1	column-2
```

Comment:
@davisusanibar I guess this column name changes occur from the Substrait end?

Reply:
Yes, it is.

I just created this issue for enhancements "Produce projection/filter extended expressions with predefined column names" https://github.com/substrait-io/substrait-java/issues/229